--- class.t3lib_tcemain.php.bak	2005-07-22 12:04:03.000000000 +0200
+++ class.t3lib_tcemain.php	2005-07-22 15:05:03.000000000 +0200
@@ -1929,6 +1929,12 @@
 		$set = true;
 
 		foreach($evalArray as $func)	{
+
+			if (substr($func,0,6)=='double'){
+				$n = substr($func,6);
+				$func = 'double';
+			}
+  
 			switch($func)	{
 				case 'int':
 				case 'year':
@@ -1938,7 +1944,7 @@
 				case 'timesec':
 					$value = intval($value);
 				break;
-				case 'double2':
+				case 'double':
 					$theDec = 0;
 					for ($a=strlen($value); $a>0; $a--)	{
 						if (substr($value,$a-1,1)=='.' || substr($value,$a-1,1)==',')	{
@@ -1947,8 +1953,12 @@
 							break;
 						}
 					}
-					$theDec = ereg_replace('[^0-9]','',$theDec).'00';
-					$value = intval(str_replace(' ','',$value)).'.'.substr($theDec,0,2);
+					$suffix = '';
+					for ($i=0;$i<=$n;$i++){
+						$suffix.='0';
+					}
+					$theDec = ereg_replace('[^0-9]','',$theDec).$suffix;
+					$value = intval(str_replace(' ','',$value)).'.'.substr($theDec,0,$n);
 				break;
 				case 'md5':
 					if (strlen($value)!=32){$set=false;}
