--- jsfunc.evalfield.js.bak	2005-07-22 14:17:23.000000000 +0200
+++ jsfunc.evalfield.js	2005-07-22 14:47:45.000000000 +0200
@@ -144,7 +144,7 @@
 	}
 	return outVal;
 }
-function evalFunc_parseDouble(value)	{
+function evalFunc_parseDouble(value,n)	{
 	var theVal = ''+value;
 	var dec=0;
 	if (!value)	return 0;
@@ -155,8 +155,12 @@
 			break;
 		}
 	}
-	dec = this.getNumChars(dec)+'00';
-	theVal=this.parseInt(this.noSpace(theVal))+TS.decimalSign+dec.substr(0,2);
+	var suffix='';
+	for (var i=0;i<=n;i++){
+		suffix = suffix+'0'
+	}
+	dec = this.getNumChars(dec)+suffix;
+	theVal=this.parseInt(this.noSpace(theVal))+TS.decimalSign+dec.substr(0,n);
 
 	return theVal;
 }
@@ -225,6 +229,11 @@
 	}
 }
 function evalFunc_input(type,inVal)	{
+	var n=0;
+	if (type.substr(0,6)=="double"){
+		n = type.substr(6);
+		type = "double";
+	}
 	if (type=="md5") {
 		return MD5(inVal);
 	}
@@ -234,8 +243,8 @@
 	if (type=="int") {
 		return this.parseInt(inVal);
 	}
-	if (type=="double2") {
-		return this.parseDouble(inVal);
+	if (type=="double") {
+		return this.parseDouble(inVal,n);
 	}
 
 	var today = new Date()
