Index: class.tx_commerce_pi3.php
===================================================================
--- class.tx_commerce_pi3.php	(Revision 23216)
+++ class.tx_commerce_pi3.php	(Arbeitskopie)
@@ -1378,31 +1378,9 @@
 	 *
 	 * @return unknown
 	 */
-	function getPaymentObject() {
+	function getPaymentObject($paymentType = '')	{
 		$paymentType = $this->getPaymentType();
-		$sysConfig = $GLOBALS['TYPO3_CONF_VARS']['EXTCONF'][$this->extKey]['SYSPRODUCTS']['PAYMENT'];
-		$config = $sysConfig['types'][strtolower((string)$paymentType) ];
-		$errorStr = NULL;
-
-		if (!isset($config['class'])) {
-			$errorStr[] = 'class not set!:' . $config['class'];
-		}
-		if (!file_exists($config['path'])) {
-			$errorStr[] = 'file not found!:' . $config['path'];
-		}
-
-		if (is_array($errorStr)) {
-			die('MAIN:FATAL! No payment possible because I don\'t know how to handle it! (' . implode(', ', $errorStr) . ')');
-		}
-
-		require_once ($config['path']);
-		$paymentObj = t3lib_div::makeInstance($config['class']);
-
-		if (method_exists($paymentObj, 'setStep')) {
-			$this->piVars['step'] = $paymentObj->setStep($_REQUEST, $this->piVars['step']);
-		}
-
-		return $paymentObj;
+		return parent::getPaymentObject($paymentType);
 	}
 
 
