Index: class.tx_commerce_pi3.php
===================================================================
--- class.tx_commerce_pi3.php	(Revision 23216)
+++ class.tx_commerce_pi3.php	(Arbeitskopie)
@@ -1683,11 +1683,20 @@
 
 		$result = '<input id="' . $step . '-' . $fieldName . '" type="text" name="' . $this->prefixId . '[' . $step . '][' . $fieldName . ']" value="' . $value . '" ' . $maxlength;
 
-		if ($fieldConfig['readonly'] == 1) {
-			$result.= ' readonly disabled /><input type="hidden" name="' . $this->prefixId . '[' . $step . '][' . $fieldName . ']" value="' . $value . '" ' . $maxlength;
-			' />';
+		if ($fieldConfig['noPrefix'] == 1)	{
+			$result = '<input id="' . $step . '-' . $fieldName . '" type="text" name="' .$fieldName . '" value="' .$value . '" ' . $maxlength;
+			if ($fieldConfig['readonly'] == 1)	{
+				$result .= ' readonly disabled /><input type="hidden" name="' . $fieldName . '" value="' . $value . '" ' . $maxlength . ' />';
+			} else {
+				$result .= '/>';
+			}
 		} else {
-			$result.= '/>';
+			$result = '<input id="' . $step . '-' . $fieldName . '" type="text" name="' . $this->prefixId . '[' . $step . '][' . $fieldName .']" value="' . $value .'" ' . $maxlength;
+			if ($fieldConfig['readonly'] == 1)	{
+				$result .= ' readonly disabled /><input type="hidden" name="' . $this->prefixId . '[' . $step . '][' . $fieldName .']" value="' . $value .'" ' . $maxlength . ' />';
+			} else {
+				$result .= '/>';
+			}
 		}
 
 		return $result;
